### PR TITLE
Added unit and e2e tests for the LicenseCode.vue component

### LINKED ISSUE #117: Add unit and e2e tests for the LicenseCode component
Unit and e2e tests need to be written for the LicenseCode component. Unit tests are done with [Jest](https://jestjs.io/), and e2e tests are done with [nightwatch](https://nightwatchjs.org/).

Please remember to test the following things:
- That individual parts of the component are present when appropriate. (unit and e2e)
- That any computed props and methods work properly, if there are any. (unit)
- Any common interactions between the user and component, if there are any. (e2e)
- Any other functionality that is unique to the component being tested!

### Additional Context
- [./src/components/LicenseCode.vue](https://github.com/creativecommons/cc-chooser/blob/master/src/components/LicenseCode.vue)
- [This repo's testing README](https://github.com/creativecommons/cc-chooser/blob/master/tests/README.md)
- [Vue's guide on unit testing](https://vuejs.org/v2/guide/unit-testing.html)
- [Vue's guide on unit testing with VueX](https://vue-test-utils.vuejs.org/guides/using-with-vuex.html)

### PR DESCRIPTION
**Pull request by hemanth-hk**
Fixes #117

## Description
This `PR` consists of the `unit` and `e2e` tests for the `LicenseCode.vue` component
`unit` tests also have the `snapshot` tests.

## Checklist
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository.
- [x] My commit messages follow [best practices][best_practices].
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and, I
    have the right to submit it under the open-source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open-source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

### CODE REVIEW

File: tests/e2e/specs/LicenseCode.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            .assert.elementPresent('.pagination-next')
+            .click('.pagination-next')
+            .assert.elementPresent('.select > select')
+            .click('.select > select')
+            .click('option[value="CC BY-SA 4.0"]')
+            .assert.elementPresent('.pagination-next')
+            .click('.pagination-next')
+            .assert.elementPresent('p[class="license-text"]')
+    },
+    'Check if the a tag in the "Rich Text" redirects to https://creativecommons.org/licenses/by-sa/4.0/?ref=ccchooser': function(browser) {
```
Comment (akmadian) [inline tests/e2e/specs/LicenseCode.js]: I think that a better name for this test would be something like "Check if the link to license code redirects properly".

File: tests/unit/specs/components/LicenseCode.spec.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    it('Check if the creatorProfileUrl function returns the correct text', () => {
+        expect(wrapper.vm.creatorProfileUrl).toBe('www.author.com')
+    })
+    it('Check if the workTitle function returns the correct text', () => {
+        expect(wrapper.vm.workTitle).toBe('My work')
+    })
+    it('Check if the workUrl function returns the correct text', () => {
+        expect(wrapper.vm.workUrl).toBe('www.author.com/pic.jpg')
+    })
+    it('Check if the isWeb function returns the correct boolean', () => {
```
Comment (akmadian) [inline tests/unit/specs/components/LicenseCode.spec.js]: Please create a test to verify that the print attribution is generated properly

File: tests/e2e/specs/LicenseCode.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+module.exports = {
+
+    'Check if the main p tag with the class license-text is present': function(browser) {
```
Comment (akmadian) [inline tests/e2e/specs/LicenseCode.js]: Please add tests to assert that the HTML and print attributions are also present.

File: tests/e2e/specs/LicenseCode.js (hunk 1 above)
Comment (hemanth-hk) [reply tests/e2e/specs/LicenseCode.js]: Done

File: tests/unit/specs/components/LicenseCode.spec.js (hunk 2 above)
Comment (hemanth-hk) [reply tests/unit/specs/components/LicenseCode.spec.js]: I added it

File: tests/e2e/specs/LicenseCode.js (hunk 3 above)
Comment (hemanth-hk) [reply tests/e2e/specs/LicenseCode.js]: Yup, Done

File: tests/unit/specs/components/LicenseCode.spec.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+import { mount, createLocalVue, config } from '@vue/test-utils'
+import LicenseCode from '@/components/LicenseCode.vue'
+import VueI18n from 'vue-i18n'
```
[the remark this answers is not in the document]
Comment (hemanth-hk) [reply tests/unit/specs/components/LicenseCode.spec.js]: I searched on the internet about this warning and found that this occurs when the translation is not found and the `i18n` is using `fallbackLocale` which in our case is `en`. Since we are testing for `en` only there should not be any issues with them
I suppressed then with an attribute

File: tests/e2e/specs/LicenseCode.js [hunk 5]
```diff
@@ -1,6 +1,6 @@
 module.exports = {
```
Comment (akmadian) [inline tests/e2e/specs/LicenseCode.js]: I would like to see some tests in here that check to verify that the HTML, richtext, and print attributions are being generated properly and have the correct values. For example, if CC BY-NC is selected, I'd like to see a check that the print attribution is of the value `This work is licensed under CC BY-NC 4.0. To view a copy of this license, visit https://creativecommons.org/licenses/by-nc/4.0`.

**Comment by akmadian** [comment]
I also just noticed that there's an uncovered line, please add a test for line 80 to be sure that the empty string there is returned when appropriate :)

File: tests/e2e/specs/LicenseCode.js (hunk 5 above)
Comment (hemanth-hk) [reply tests/e2e/specs/LicenseCode.js]: Ya, I added them, there were many span tags so I had to add a test for each of them... but done:)

**Comment by hemanth-hk** [comment]
Thank you for reviewing the code even though you were busy...I made all the changes, hence review it whenever you are free :)

File: tests/e2e/specs/LicenseCode.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    'Check if the p-tag with the class license-text is present': function(browser) {
+        browser
+            .init()
+            .assert.elementPresent('.control-label > span')
+            .click('.control-label > span')
+            .assert.elementPresent('.pagination-next')
+            .click('.pagination-next')
+            .assert.elementPresent('.select > select')
+            .click('.select > select')
```
Comment (obulat) [inline tests/e2e/specs/LicenseCode.js:11]: This line causes an error running tests:

> Error while running .clickElement() protocol action: element click intercepted: Element <select>...</select> is not clickable at point (871, 102). Other element would receive the click
: <span class="select">...</span>

If you click on span.select, you will get the result you were expecting, and the tests run fine.

The same happens in LicenseDetailsCard

Comment (hemanth-hk) [reply tests/e2e/specs/LicenseCode.js:11]: This test passed when I was sending a pull request, thank you for pointing this out I'll surely edit this :)

Comment (obulat) [reply tests/e2e/specs/LicenseCode.js:11]: > This test passed when I was sending a pull request, thank you for pointing this out I'll surely edit this :)

I think it passed for me before, too. Anyways, I opened an issue, and I think it's best if I assign it to you. Thank you for quick reply!
